### ISSUE TITLE
[BUG] panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by wangxiaoq** [koordinator-sh/koordinator#443]
**What happened**:

I deploy koordinator in a K8s cluster, and one node start koord-runtime-proxy failed with the following error messages, other nodes are ok:

```
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: panic: runtime error: invalid memory address or nil pointer dereference
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x17298db]
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: goroutine 151 [running]:
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: github.com/koordinator-sh/koordinator/pkg/runtimeproxy/resexecutor/cri.(*ContainerResourceExecutor).ResourceCheckPoint(0xc000416fd0, 0x1af4b80, 0xc000a5ff10, 0xc000d151e0, 0x0)
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]:         /root/projects/koordinator/pkg/runtimeproxy/resexecutor/cri/container.go:135 +0x7b
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: github.com/koordinator-sh/koordinator/pkg/runtimeproxy/server/cri.(*RuntimeManagerCriServer).failOver(0xc000536330, 0x7ffec00c1eda, 0x23)
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]:         /root/projects/koordinator/pkg/runtimeproxy/server/cri/criserver.go:187 +0x2de
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: github.com/koordinator-sh/koordinator/pkg/runtimeproxy/server/cri.(*RuntimeManagerCriServer).Run(0xc000536330, 0x0, 0x0)
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]:         /root/projects/koordinator/pkg/runtimeproxy/server/cri/criserver.go:60 +0x97
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]: created by main.main
Aug 04 17:17:55 xx.xx.xx.xx koord-runtime-proxy-0.5.0[580852]:         /root/projects/koordinator/cmd/koord-runtime-proxy/main.go:57 +0x59b

```

**What you expected to happen**:

koord-runtime-proxy start successfully

**How to reproduce it (as minimally and precisely as possible)**:

Just install koordinator on a node.

**Anything else we need to know?**:

**Environment**: 
- App version: 0.5.0
- Kubernetes version (use `kubectl version`): 1.18.6
- Install details (e.g. helm install args): `--set featureGates="PodMutatingWebhook=true\,PodValidatingWebhook=true"`
- Others:

**Comment by wangxiaoq** [comment]
more info:

I found errors in kubelet logs:

```
kubelet-1.18.6[132718]: F0804 17:46:38.889050  132718 kubelet.go:1384] Failed to start ContainerManager failed to build map of initial containers from runtime: no PodsandBox found with Id 'b533e8e0efb6d6b2231488c57f2f83b1125935293fcfb903b9e8b109256688d6'
```

`b533e8e0efb6d6b2231488c57f2f83b1125935293fcfb903b9e8b109256688d6` is an exited containers, and after the deletion of this container, the koord-runtime-proxy start successfully.

If you think it's not a problem, please close it, thank you!

**Comment by hormes** [comment]
docker or containerd environment？

**Comment by wangxiaoq** [comment]
> docker or containerd environment？

containerd 1.4.11

**Comment by cheimu** [comment]
Hi @wangxiaoq , could you try v0.6.0 or the latest code? Since this may be fixed in pr https://github.com/koordinator-sh/koordinator/pull/355 which is released in v0.6.0
